core/src/main/java/org/bitcoinj/params/MainNetParams.java

```diff
@@ -31,11 +32,15 @@
     public static final int MAINNET_MAJORITY_WINDOW = 1000;
     public static final int MAINNET_MAJORITY_REJECT_BLOCK_OUTDATED = 950;
     public static final int MAINNET_MAJORITY_ENFORCE_BLOCK_UPGRADE = 750;
+    private static final long DIFFICULTY_TARGET = 0x1d00ffffL;
```

Comment:
Can the introduction of constants perhaps be extracted into a first commit?

Reply:
Yes. This one is definitely a candidate for independent review. I think there is still some duplication that can be eliminated. I also don't understand the full behavior and rationale for the various difficulty targets and limits. So if you can help me add some comments that would be great.